java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -336,7 +339,21 @@ class DatabaseImpl(
                 } else {
                     BigInteger(it.getString(4)).toReferencable()
                 }
-                else -> if (it.isNull(6)) {
+                else -> if (isCollection == FieldClass.InlineEntity) {
+                    val rawSingletons = mutableMapOf<FieldName, Referencable?>()
+                    val rawCollections = mutableMapOf<FieldName, Set<Referencable>>()
+                    val (dbSingletons, dbCollections) =
+                        getEntityFields(it.getLong(3), counters, db)
+                    dbSingletons.forEach { (fieldName, value) -> rawSingletons[fieldName] = value }
+                    dbCollections.forEach {
+                        (fieldName, value) -> rawCollections[fieldName] = value
+                    }
+                    RawEntity(
+                        id = "",
```

Comment:
I think you are writing the id on the way in, but not returning it on the way back? maybe add a comment here explaining why it is ok to have it blank

Reply:
might be best to always write an empty id too, wdyt? I could also assert that the id provided was empty to make sure programming errors don't creep in